tests/wrappers/test_autoreset.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+from typing import Optional
+
+import numpy as np
+
+import gym
+from gym.wrappers import AutoResetWrapper
+
+
+class DummyResetEnv(gym.Env):
+    metadata = {}
+
+    def __init__(self):
+        self.action_space = gym.spaces.Box(low=np.array([-1.0]), high=np.array([1.0]))
+        self.observation_space = gym.spaces.Box(
+            low=np.array([-1.0]), high=np.array([1.0])
+        )
+        self.count = 0
+
+    def step(self, action):
+        self.count += 1
+        return (
+            np.array([self.count]),
+            1 if self.count > 2 else 0,
+            self.count > 2,
+            {"count": self.count},
+        )
+
+    def reset(
+        self,
+        *,
+        seed: Optional[int] = None,
+        return_info: Optional[bool] = False,
+        options: Optional[dict] = None
+    ):
+        self.count = 0
+        if not return_info:
+            return np.array([self.count])
+        else:
+            return np.array([self.count]), {"count": self.count}
+
+
+def test_autoreset_reset_info():
+    env = gym.make("CartPole-v1")
```

Comment:
You can parametrize the tests to run for all environments, the pattern appears in some other tests

Reply:
@RedTachyon  The problem with testing the wrapper for all environments is for the real environments often they have non-deterministic starting state, and non-deterministic transitions, meaning it is difficult to know what the correct return values should be for various cases, and when done will be true, without setting a seed and then recording the behavior of each environment and hard coding a test for the recorded behavior. That was my motivation for creating the deterministic dummy environment.

Reply:
I didn't do that for the autoreset wrapper since a lot of the other wrappers are not tested with all environments. My thinking is that if we test every wrapper with every first party environment, it could make the tests take a lot longer.